### PR TITLE
Account list redesign

### PR DESCRIPTION
**Pull request by 12people**
Pushing a the redesign of the account list.

In the process of doing the redesign, I discovered that a number of the logos blatantly went against companies' guidelines on when to use their logos, which makes me think they were used without permission. In order to stay safer legally, I replaced the offending logos with simple avatars—their first letters inside circles.

The "Show disabled accounts" button has been moved to the bottom for better discoverability. The grid and table views were merged into one view featuring the name of the service as well as switches for turning upload or social on or off.

screenshot_20180424-010217

### CODE REVIEW

**Comment by 12people** [comment]
@gerhardol This is the Froyo-compatible patch I was talking about.

If any future commit requires API 14 or above, I'm okay with raising the minSDK. If it doesn't, though, I'd like to support Froyo as long as possible (since there are still a few people using it).

One thing that I'd really like to change—could we raise the compileSDK to 27?

**Comment by 12people** [comment]
@gerhardol 
I have rebased before submitting and the conflicts listed are all very one-sided. Perhaps these specific files are locked somehow? As you have write access, could you resolve these conflicts?

**Comment by 12people** [comment]
As for Froyo—if it's limiting you, I'm absolutely fine if you raise the minSDK to 14 after this particular commit is accepted.

**Comment by gerhardol** [comment]
Please rebase on masterand straighten out the merge commits in this branch to simplify reviewing.

**Comment by 12people** [comment]
@gerhardol 
I'll try to rebase again, though as I said—this should be rebased based on the current master here.

BTW, I keep running into this problem: whenever I try to rebase or pull or anything, git keeps telling me I need to commit changes to gradlew.bat . Have you run into this problem? What should I do? How should I avoid it?

**Comment by gerhardol** [comment]
The PR was not based on master before. For sure not now when I have pushed some changes, including Android-27 (not much tested in current master though). A few more fixes in the pipe.
I will prefer if the PR was cleaned up, then I can merge it more easily.

For gradlew.bat: .gitattributes forces the file to crlf line endings. An old clone may not force that.
Try
$ unix2dos gradlew.bat

Thanks for the contributions. Sorry that I do not have more time right now.

**Comment by 12people** [comment]
@gerhardol Just rebased this branch.
Let me know what you think.

**Comment by 12people** [comment]
Alright, no problem, take your time.

File: app/src/org/runnerup/view/AccountListActivity.java [hunk 1]
```diff
@@ -51,35 +54,60 @@
...
     CursorAdapter mCursorAdapter;
 
-    /** Called when the activity is first created. */
+    /**
+     * Called when the activity is first created.
+     */
 
     @Override
-    public void onCreate(Bundle savedInstanceState) {
+    public void onCreate(Bundle savedInstanceState) { //todo write up logo usage permissions
```
Comment (gerhardol) [inline app/src/org/runnerup/view/AccountListActivity.java]: This comment should not be needed

File: app/src/org/runnerup/view/AccountListActivity.java [hunk 2]
```diff
@@ -123,118 +138,164 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
 
         return new SimpleCursorLoader(this, mDB, DB.ACCOUNT.TABLE, from, showDisabled, null,
-                DB.ACCOUNT.AUTH_CONFIG + " is null, " + DB.ACCOUNT.ENABLED + " desc, " + DB.ACCOUNT.NAME);
+                DB.ACCOUNT.AUTH_CONFIG + " is null, " + DB.ACCOUNT.ENABLED + " desc, " + DB.ACCOUNT.NAME + " collate nocase"); //todo I'm guessing AUTH_CONFIG lets me know whether an account is set up
```
Comment (gerhardol) [inline app/src/org/runnerup/view/AccountListActivity.java]: Remove this todo

File: app/src/org/runnerup/view/AccountListActivity.java [hunk 3]
```diff
@@ -123,118 +138,164 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
+                            categories[curPosition] = R.string.accounts_category_connected;
+                        }
+                    } else {
+                        // get data for previous item
+                        cursor.moveToPrevious();
+                        String prevConnected = DBHelper.get(cursor).getAsString(DB.ACCOUNT.AUTH_CONFIG);
+                        cursor.moveToNext();
+
+                        // compare the two
+                        if (prevConnected == null && curConnected != null
```
Comment (gerhardol) [inline app/src/org/runnerup/view/AccountListActivity.java]: This is not working, with two active accounts you see "Connect an account" also for second connected.

screenshot_20180603-233955

File: app/src/org/runnerup/view/AccountListActivity.java [hunk 4]
```diff
@@ -123,118 +138,164 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
 
             // upload box
             accountUploadBox.setTag(id);
-            accountUploadBox.setOnCheckedChangeListener(new CheckBox.OnCheckedChangeListener() {
+            setCustomThumb(accountUploadBox, R.drawable.switch_upload, context);
+            accountUploadBox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
                 @Override
                 public void onCheckedChanged(CompoundButton arg0, boolean arg1) {
-                    setFlag(arg0.getTag(), DB.ACCOUNT.FLAG_UPLOAD, arg1);
+                    setFlag((String) arg0.getTag(), DB.ACCOUNT.FLAG_UPLOAD, arg1); //todo doesn't seem to work
```
Comment (gerhardol) [inline app/src/org/runnerup/view/AccountListActivity.java]: Seem to work for me. What was the reason for the comment?

**Comment by 12people** [comment]
I'm in the middle of finals now—I'll get back to you soon, once I have a bit more time.

**Comment by gerhardol** [comment]
> I'm in the middle of finals now—I'll get back to you soon, once I have a bit more time.

Good luck!

**Comment by 12people** [comment]
@gerhardol I fixed the bug as well as pulled in changes from master.

BTW, I still haven't figured out how to effectively deal with line endings in gradlew. This time, I removed the offending line from .gitattributes, stashed, then commited, but that's not very effective.

**Comment by gerhardol** [comment]
Compile error (Travis and for me). I cannot see why right now.

**Comment by 12people** [comment]
@gerhardol It seems that, when pulling in changes from master, a curly brace accidentally got lost. It should work now, I hope.

File: app/res/values/styles.xml [hunk 5]
```diff
@@ -84,4 +84,14 @@
         <item name="android:textColor">@color/bottom_tabs</item>
     </style>
 
+    <style name="AccountListText">
+        <item name="android:textSize">18sp</item>
+    </style>
+
+    <style name="ListCategoryHeading">
+        <item name="android:fontFamily">@string/fontFamily_medium</item>
```
Comment (gerhardol) [inline app/res/values/styles.xml]: Separate styles-16 needed

F:\dev\gc\runnerup\app\res\values\styles.xml:92: Error: android:fontFamily requires API level 16 (current min is 8) [NewApi]
        <item name="android:fontFamily">@string/fontFamily_medium</item>

**Comment by gerhardol** [comment]
Thanks for the update. Will you leave it at this or should I wait before merging?

**Comment by 12people** [comment]
You may merge it. :)

**Comment by 12people** [comment]
And after you do and make a release for Froyo, it's okay with me if the minAPI is increased to 14 or 15.

**Comment by gerhardol** [comment]
@12people 
(Can open issues about this but trying to continue discussion in this PR)

This branch has introduced some lints in the build. I intend to merge master_ng eventually that eliminates lints. Can you check that the issues are addressed correctly?
(it seems to me that the background painting is OK to remove)
(There are two GradleDepreciation too, will suppress those, I believe at least Play was due to Froyo).

https://github.com/gerhardol/runnerup/commit/e778301ed4c05b4da4382634ff3336e016ff03f9

F:\dev\gc\runnerup\app\res\drawable-hdpi\title_spinner_selected.9.png: Warning: The following unrelated icon files have identical contents: title_spinner_pressed.9.png, title_spinner_selected.9.png [IconDuplicates]
Also affects:
     F:\dev\gc\runnerup\app\res\drawable-hdpi\title_spinner_pressed.9.png

:app:javaPreCompileLatestDebug
F:\dev\gc\runnerup\app\src\org\runnerup\view\AccountListActivity.java:94: warning: [unchecked] unchecked cast
                getSupportLoaderManager().restartLoader(0, null, (LoaderCallbacks<Object>) view.getContext());
                                                                                                          ^
  required: LoaderCallbacks<Object>
  found:    Context

Text may be too long - remove the labels?
screenshot_20180625-002137

**Comment by gerhardol** [comment]
For the identical images, we can suppress the warning.

Did you take a look of the corrections I committed, are the functionality as expected?

**Comment by gerhardol** [comment]
account_row.xml : Will revert and suppress

bottom_tab_indicator.xml : Yes, that is what it looked like, just wanted to know that it was intentional

**Comment by 12people** [comment]
About the bottom tab: I don't remember. As long as it also works on Froyo, it's good. :)

**Pull request merged**